### PR TITLE
fixed php-amqplib composer version

### PR DESCRIPTION
**Pull request by a-sabaa**
- Developer(s): @a-sabaa

- Ticket: TICKET_URL_HERE

- Academy PR: ACADEMY_URL_HERE

- Release Group: https://release.spryker.com/release-groups/view/1470

#### Please confirm

- [ ] No new OS components - or they have been approved by legal department

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   RabbitMq              | minor                 |                       |

#### Release Notes

{skip}

-----------------------------------------

##### Change log

Improvements

- Added all `PhpAmqpLib\Connection\AMQPStreamConnection` constructor parameter defaults to class creation in factory
- Added all constructor parameters to `Generated\Shared\Transfer\QueueConnectionTransfer`
- Defaulted HeartBeat for `PhpAmqpLib\Connection\AMQPStreamConnection` creation to `0`

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "license": "MIT",
   "require": {
     "php": ">=7.1",
-    "php-amqplib/php-amqplib": "^2.6",
+    "php-amqplib/php-amqplib": "2.9.0",
```
Comment (dereuromark) [inline composer.json]: Cant we use ~ here at least?

Comment (a-sabaa) [reply composer.json]: This is tmp until 2.10, I am not sure back-ported patches would be that necessary to worry about..also 2.9.1 is the problem..which would be grabbed by ~

**Comment by dereuromark** [comment]
template missing - before this can be set to ready for release.

File: composer.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
   "license": "MIT",
   "require": {
     "php": ">=7.1",
-    "php-amqplib/php-amqplib": "2.9.0",
+    "php-amqplib/php-amqplib": "^2.6",
```
Comment (dereuromark) [inline composer.json]: ^ is pulling 2.7 and 2.8 and 2.9?

Comment (a-sabaa) [reply composer.json]: Yes, we overwrote the problem on core level, now it's compatible with whatever "BC" stuff they throw at us when it comes to heartbeat

File: src/Spryker/Client/RabbitMq/Model/Connection/ConnectionFactory.php [hunk 3]
```diff
@@ -40,18 +40,29 @@ public function getQueueConnectionConfigs()
...
+            $queueConnectionTransfer->getUsername(),
+            $queueConnectionTransfer->getPassword(),
+            $queueConnectionTransfer->getVirtualHost(),
+            $queueConnectionTransfer->getInsist() ?? false,
+            $queueConnectionTransfer->getLoginMethod() ?? 'AMQPLAIN',
+            $queueConnectionTransfer->getLoginResponse(),
+            $queueConnectionTransfer->getLocale() ?? 'en_US',
+            $queueConnectionTransfer->getConnectionTimeout() ?? 3,
+            $queueConnectionTransfer->getReadWriteTimeout() ?? 130,
+            null,
```
Comment (ehsanmx) [inline src/Spryker/Client/RabbitMq/Model/Connection/ConnectionFactory.php]: is this always null?
vendor/php-amqplib/php-amqplib/PhpAmqpLib/Wire/IO/StreamIO.php:59

File: src/Spryker/Shared/RabbitMq/Transfer/rabbitmq.transfer.xml [hunk 4]
```diff
@@ -6,11 +6,23 @@
 
     <transfer name="QueueConnection">
         <property name="name" type="string" />
+
```
Comment (ehsanmx) [inline src/Spryker/Shared/RabbitMq/Transfer/rabbitmq.transfer.xml]: why empty line?

File: src/Spryker/Client/RabbitMq/Model/Connection/ConnectionFactory.php (hunk 3 above)
Comment (ehsanmx) [reply src/Spryker/Client/RabbitMq/Model/Connection/ConnectionFactory.php]: This looks like `$login_response`

Comment (a-sabaa) [reply src/Spryker/Client/RabbitMq/Model/Connection/ConnectionFactory.php]: it's not always null, but we cannot have a "resource" in a transfer object

File: src/Spryker/Shared/RabbitMq/Transfer/rabbitmq.transfer.xml (hunk 4 above)
Comment (a-sabaa) [reply src/Spryker/Shared/RabbitMq/Transfer/rabbitmq.transfer.xml]: just to separate the "business" properties from the "required to create connection" properties

**Pull request merged**